bot/exts/utilities/color.py

```diff
@@ -0,0 +1,229 @@
+        hex_colour = self.match_colour_name(user_colour_name)
+        hex_tuple = ImageColor.getrgb(hex_colour)
+        await self.send_colour_response(ctx, hex_tuple)
+
+    @colour.command()
+    async def random(self, ctx: commands.Context) -> None:
+        """Command to create an embed from a randomly chosen colour from the reference file."""
+        hex_colour = random.choice(list(self.colour_mapping.values()))
+        hex_tuple = ImageColor.getrgb(f"#{hex_colour}")
+        await self.send_colour_response(ctx, hex_tuple)
+
+    def get_colour_conversions(self, rgb: tuple[int, int, int]) -> dict[str, str]:
+        """Create a dictionary mapping of colour types and their values."""
+        return {
+            "RGB": rgb,
+            "HSV": self._rgb_to_hsv(rgb),
+            "HSL": self._rgb_to_hsl(rgb),
+            "CMYK": self._rgb_to_cmyk(rgb),
+            "Hex": self._rgb_to_hex(rgb),
+            "Name": self._rgb_to_name(rgb)
+        }
+
+    @staticmethod
+    def _rgb_to_hsv(rgb: tuple[int, int, int]) -> tuple[int, int, int]:
+        """Convert RGB values to HSV values."""
+        rgb_list = [val / 255 for val in rgb]
+        h, s, v = colorsys.rgb_to_hsv(*rgb_list)
+        hsv = (round(h * 360), round(s * 100), round(v * 100))
+        return hsv
+
+    @staticmethod
+    def _rgb_to_hsl(rgb: tuple[int, int, int]) -> tuple[int, int, int]:
+        """Convert RGB values to HSL values."""
+        rgb_list = [val / 255.0 for val in rgb]
+        h, l, s = colorsys.rgb_to_hls(*rgb_list)
+        hsl = (round(h * 360), round(s * 100), round(l * 100))
+        return hsl
+
+    @staticmethod
+    def _rgb_to_cmyk(rgb: tuple[int, int, int, int]) -> tuple[int, int, int, int]:
+        """Convert RGB values to CMYK values."""
+        rgb_list = [val / 255.0 for val in rgb]
+        if not any(rgb_list):
+            return 0, 0, 0, 100
+        k = 1 - max(val for val in rgb_list)
+        c = round((1 - rgb_list[0] - k) * 100 / (1 - k))
+        m = round((1 - rgb_list[1] - k) * 100 / (1 - k))
+        y = round((1 - rgb_list[2] - k) * 100 / (1 - k))
+        cmyk = (c, m, y, round(k * 100))
+        return cmyk
+
+    @staticmethod
+    def _rgb_to_hex(rgb: tuple[int, int, int]) -> str:
+        """Convert RGB values to HEX code."""
+        hex_ = ''.join([hex(val)[2:].zfill(2) for val in rgb])
+        hex_code = f"#{hex_}".upper()
+        return hex_code
+
+    def _rgb_to_name(self, rgb: tuple[int, int, int]) -> str:
+        """Convert RGB values to a fuzzy matched name."""
+        input_hex_colour = self._rgb_to_hex(rgb)
+        try:
+            match, certainty, _ = process.extractOne(
+                query=input_hex_colour,
+                choices=self.colour_mapping.values(),
+                score_cutoff=80
+            )
+            colour_name = [name for name, hex_code in self.colour_mapping.items() if hex_code == match][0]
+        except TypeError:
+            colour_name = "No match found"
```

Comment:
Rather than no match found, returning None and dealing with it elsewhere should be done.

This function reads like... The return result is a colour name, unless the name is 'No match found' then there's no colour. None is the python type for this.